lib/plugins/ramdisk.js

```diff
@@ -10,16 +10,34 @@
 */
 /* eslint-disable no-param-reassign */
 const crypto = require('crypto');
-const { symlinkSync } = require('fs');
+const { symlinkSync, readFileSync } = require('fs');
 const { basename, join, resolve } = require('path');
 
 const isCwd = require('is-path-cwd');
-const readPkgUp = require('read-pkg-up');
+const escalade = require('escalade/sync');
 const rm = require('rimraf');
 const { WebpackPluginRamdisk } = require('webpack-plugin-ramdisk');
 
 const { PluginExistsError, RamdiskPathError } = require('../errors');
 
+const readPkgName = () => {
+  const pkgPath = escalade(process.cwd(), (dir, names) => {
+    if (names.includes('package.json')) {
+      return 'package.json';
+    }
+    return false;
+  });
+  if (pkgPath == null) {
+    return null;
+  }
+  try {
+    const pkg = JSON.parse(readFileSync(pkgPath, 'utf-8'));
+    return pkg.name;
+  } catch (error) {
+    return null;
+  }
+};
+
```

Comment:
we're going to need test coverage of this method before we can merge

Reply:
As I mentioned above. This is covered by existing tests. See https://github.com/shellscape/webpack-plugin-serve/pull/187/files#diff-817a4e640cdcf9964df2158c8020f627R33-R54

Though I can add test with  broken package.json. This way random name can be tested too.

Reply:
No, broken package.json will not work because webpack-nano failed as well for some reason.